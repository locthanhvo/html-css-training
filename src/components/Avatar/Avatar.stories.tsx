import { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import Avatar from '.';

export default {
  title: 'Components/Avatar',
  component: Avatar,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
} as Meta;

type Story = StoryObj<typeof Avatar>;

export const AvatarDefault: Story = {
  args: {
    firstName: 'John',
    lastName: 'Doe',
    email: 'HbJjv@example.com',
  },
};
