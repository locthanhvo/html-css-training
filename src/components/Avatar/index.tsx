import { Center, Flex, Image, Text } from '@chakra-ui/react';

import AvatarDefault from '../../assets/images/avatar.png';
import { memo } from 'react';

interface AvatarProps {
  url?: string;
  firstName: string;
  lastName: string;
  email: string;
}

const Avatar = ({ url, firstName, lastName, email }: AvatarProps) => {
  return (
    <Flex gap={2} alignItems="center">
      <Center w={10} h={10} borderRadius="50%">
        <Image src={url || AvatarDefault} />
      </Center>

      <Flex direction="column">
        <Text color="primary" fontWeight="medium" fontSize="base">
          {firstName} {lastName}
        </Text>
        <Text color="primary" fontWeight="regular" fontSize="xs">
          {email}
        </Text>
      </Flex>
    </Flex>
  );
};

export default memo(Avatar);
