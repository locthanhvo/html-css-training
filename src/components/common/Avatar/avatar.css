.avatar-item {
  width: 38px;
  height: 38px;
  border-radius: 50%;
}

.avatar-name {
  background-color: var(--color-purple);
  color: var(--color-white);
  font-size: var(--font-size-2sm);
}
