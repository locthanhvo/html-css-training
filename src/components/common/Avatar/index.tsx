import { Center, Image } from '@chakra-ui/react';
import { memo } from 'react';

interface AvatarProps {
  url?: string;
  w?: number;
  h?: number;
}

const Avatar = ({ url, w = 20, h = 20 }: AvatarProps) => {
  return (
    <Center w={w} h={h} borderRadius="50%">
      <Image borderRadius="50%" src={url} />
    </Center>
  );
};

export default memo(Avatar);
