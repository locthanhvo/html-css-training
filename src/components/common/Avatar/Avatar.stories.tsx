import type { Meta, StoryObj } from "@storybook/react";
import Avatar from ".";
import "../../../index.css";

const meta = {
  title: "Components/Avatar",
  component: Avatar,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Avatar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const AvatarNameUser: Story = {
  args: {
    firstName: "Loc",
    lastName: "Vo",
  },
};

export const AvatarImageUser: Story = {
  args: {
    url: "https://png.pngtree.com/png-vector/20220709/ourmid/pngtree-businessman-user-avatar-wearing-suit-with-red-tie-png-image_5809521.png",
    name: "Loc Vo",
    firstName: "Loc",
    lastName: "Vo",
  },
};
