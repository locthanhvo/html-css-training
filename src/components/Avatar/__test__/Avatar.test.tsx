import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import Avatar from '..';

describe('Avatar Component', () => {
  it('Should match snapshot', () => {
    const element = render(
      <BrowserRouter>
        <Avatar firstName="John" lastName="Doe" email="HbJjv@example.com" />
      </BrowserRouter>,
    );

    expect(element).toMatchSnapshot();
  });
});
