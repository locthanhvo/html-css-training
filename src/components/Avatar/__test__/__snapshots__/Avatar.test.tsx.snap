// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-r2hvuq"
      >
        <div
          class="css-1x4sxyt"
        >
          <img
            class="chakra-image css-0"
            src="test-file-stub"
          />
        </div>
        <div
          class="css-j7qwjs"
        >
          <p
            class="chakra-text css-1j7zc5v"
          >
            John
             
            Doe
          </p>
          <p
            class="chakra-text css-z171l4"
          >
            HbJjv@example.com
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-r2hvuq"
    >
      <div
        class="css-1x4sxyt"
      >
        <img
          class="chakra-image css-0"
          src="test-file-stub"
        />
      </div>
      <div
        class="css-j7qwjs"
      >
        <p
          class="chakra-text css-1j7zc5v"
        >
          John
           
          Doe
        </p>
        <p
          class="chakra-text css-z171l4"
        >
          HbJjv@example.com
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
